### ISSUE TITLE
Helm: Helm vs Git - configuration polish

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-cd#2259]
**Is your feature request related to a problem? Please describe.**

Currently we have Git fields on `Repository` and have `Type` as a type-descrimitator. We could instead have a `Git` and a `Helm` field.

**Comment by alexec** [comment]
@alexmt @jessesuen not sure about this one. I don't think we should do this. It breaks backward compatibility for Git.

**Comment by alexmt** [comment]
I disagree. The `helm.repositories` is also breaking change. Since we decided to merge  `helm.repositories`  and  `repositories`  we should do it right. Support `helm.repositories` and `repositories` git only fields, depreated them and eventually drop.

**Comment by alexec** [comment]
@alexmt can we close this?

**Comment by alexmt** [comment]
I would prefer to work on it in v1.4.

**Comment by alexec** [comment]
Move to v1.4

**Comment by andrii-korotkov-verkada** [comment]
Closed since there's been quite a long time since updates and it's unclear if this is still relevant. Feel free to re-open/re-submit the issue if still relevant.
